hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/util/curator/TestSecureZKCuratorManager.java

```diff
@@ -0,0 +1,157 @@
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.hadoop.fs.FileContext.LOG;
+import static org.junit.Assert.assertEquals;
+
+
+/**
+ * Test the manager for ZooKeeper Curator when SSL/TLS is enabled for the ZK server-client connection.
+ */
+public class TestSecureZKCuratorManager {
+
+  private TestingServer server;
+  private ZKCuratorManager curator;
+  private Configuration hadoopConf;
+  private Integer secureClientPort = 2281;
+  private File zkDataDir = new File("testZkSSLClientConnectionDataDir");
+
+  @Before
+  public void setup() throws Exception {
+    //set zkServer
+    this.hadoopConf = setUpSecure();
+    Map<String, Object> customConfiguration = new HashMap<>();
+    customConfiguration.put("secureClientPort",this.secureClientPort.toString());
+    customConfiguration.put("audit.enable",true);
+
+    InstanceSpec spec = new InstanceSpec(
+            this.zkDataDir,
+            this.secureClientPort,
+            -1,
+            -1,
+            true,
+            1,
+            100,
+            10,
+            customConfiguration);
+    this.server = new TestingServer(spec, true);
+    hadoopConf.set(CommonConfigurationKeys.ZK_ADDRESS, this.server.getConnectString());
+    this.curator = new ZKCuratorManager(hadoopConf);
+    this.curator.start(new ArrayList<>(), true);
+  }
+
+  public Configuration setUpSecure() throws Exception {
+    Configuration hadoopConf = new Configuration();
+    String testDataPath = "src/test/java/org/apache/hadoop/util/curator/resources/data";
+    System.setProperty("zookeeper.serverCnxnFactory", "org.apache.zookeeper.server.NettyServerCnxnFactory");
+    //System.setProperty("zookeeper.client.secure", "true");
+
+
+    System.setProperty("zookeeper.ssl.keyStore.location", testDataPath + "/ssl/keystore.jks");
+    System.setProperty("zookeeper.ssl.keyStore.password", "password");
+    System.setProperty("zookeeper.ssl.trustStore.location", testDataPath + "/ssl/truststore.jks");
+    System.setProperty("zookeeper.ssl.trustStore.password", "password");
+    System.setProperty("zookeeper.request.timeout", "12345");
+
+    System.setProperty("jute.maxbuffer", "469296129");
+
+    System.setProperty("javax.net.debug", "ssl");
+    System.setProperty("zookeeper.authProvider.x509", "org.apache.zookeeper.server.auth.X509AuthenticationProvider");
+
+
+    hadoopConf.set(CommonConfigurationKeys.ZK_SSL_KEYSTORE_LOCATION, testDataPath + "/ssl/keystore.jks");
+    hadoopConf.set(CommonConfigurationKeys.ZK_SSL_KEYSTORE_PASSWORD, "password");
+    hadoopConf.set(CommonConfigurationKeys.ZK_SSL_TRUSTSTORE_LOCATION, testDataPath + "/ssl/truststore.jks");
+    hadoopConf.set(CommonConfigurationKeys.ZK_SSL_TRUSTSTORE_PASSWORD, "password");
+    return hadoopConf;
+  }
+
+  @After
+  public void teardown() throws Exception {
+    this.curator.close();
+    if (this.server != null) {
+      this.server.close();
+      this.server = null;
```

Comment:
what's the reason of setting it to null?

Reply:
As the setup is annotated with @Before, it always initializes this.server with an Object. I don't think setting it to null makes anything better. Let's leave it as it is, though.